diff -ruN v8-0.40.2.orig/build.rs v8-0.40.2/build.rs
--- v8-0.40.2.orig/build.rs	2022-03-31 23:47:04.516655587
+++ v8-0.40.2/build.rs	2022-03-31 23:49:25.154897367
@@ -55,7 +55,22 @@
 
   if !(is_trybuild || is_cargo_doc | is_rls) {
     if env::var_os("V8_FROM_SOURCE").is_some() {
-      build_v8()
+      // utilize a lockfile to prevent building v8 twice
+      let root = env::current_dir().unwrap();
+      let out_dir = env::var_os("OUT_DIR").unwrap();
+      let lockfilepath = root
+        .join(out_dir)
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .join("building_v8.fslock");
+      println!("building v8 lockfile: {:?}", &lockfilepath);
+      let mut lockfile = LockFile::open(&lockfilepath)
+        .expect("Couldn't open lib download lockfile.");
+      lockfile.lock().expect("Couldn't get lock");
+      build_v8();
+      lockfile.unlock().expect("Couldn't unlock lockfile");
     } else {
       // utilize a lockfile to prevent linking of
       // only partially downloaded static library.
