diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 62003d5..fb323f8 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2501,7 +2501,7 @@ config("minimal_symbols") {
     # flag, so we can use use -g1 for pnacl and nacl-clang compiles.
     # gcc nacl is is_nacl && !is_clang, pnacl and nacl-clang are && is_clang.
     # Windows: MSVC doesn't have an equivalent option.
-  } else if (is_win || line_tables_only) {
+  } else if ((is_win || line_tables_only) && is_clang) {
     # On Mac/Linux using Clang/LLVM 15, the -gline-tables-only flag must be the
     # **last** -gxxx flag. All other flags like -gno-inline-line-tables and
     # -gno-column-info must come first, otherwise full debuginfo is produced.
