https://github.com/rust-lang/rust-bindgen/issues/2945
--- a/clang.rs
+++ b/clang.rs
@@ -1610,7 +1610,7 @@ impl SourceLocation {
             let mut line = 0;
             let mut col = 0;
             let mut off = 0;
-            clang_getSpellingLocation(
+            clang_getFileLocation(
                 self.x, &mut file, &mut line, &mut col, &mut off,
             );
             (File { x: file }, line as usize, col as usize, off as usize)
