diff --git a/package/src/common/configure.ts b/package/src/common/configure.ts
index a8a97beb..d079fce2 100644
--- a/package/src/common/configure.ts
+++ b/package/src/common/configure.ts
@@ -41,23 +41,6 @@ export async function configure(
   info("******************************************");
   info("");
 
-  // Download dependencies
-  for (const dependency of kDependencies) {
-    try {
-      await configureDependency(dependency, config);
-    } catch (e) {
-      if (
-        e.message ===
-          "The architecture aarch64 is missing the dependency deno_dom"
-      ) {
-        info("\nIgnoring deno_dom dependency on Apple Silicon");
-        continue;
-      } else {
-        throw e;
-      }
-    }
-  }
-
   // Move the quarto script into place
   info("Creating Quarto script");
   if (Deno.build.os === "windows") {
