diff --git a/package/src/common/configure.ts b/package/src/common/configure.ts
index a8a97bebb..5bbfa1ce2 100644
--- a/package/src/common/configure.ts
+++ b/package/src/common/configure.ts
@@ -41,23 +41,6 @@ export async function configure(
   info("******************************************");
   info("");
 
-  // Download dependencies
-  for (const dependency of kDependencies) {
-    try {
-      await configureDependency(dependency, config);
-    } catch (e) {
-      if (
-        e.message ===
-          "The architecture aarch64 is missing the dependency deno_dom"
-      ) {
-        info("\nIgnoring deno_dom dependency on Apple Silicon");
-        continue;
-      } else {
-        throw e;
-      }
-    }
-  }
-
   // Move the quarto script into place
   info("Creating Quarto script");
   if (Deno.build.os === "windows") {
@@ -66,10 +49,6 @@ export async function configure(
       join(config.directoryInfo.bin, "quarto.cmd"),
     );
   } else {
-    Deno.copyFileSync(
-      join(config.directoryInfo.pkg, "scripts", "common", "quarto"),
-      join(config.directoryInfo.bin, "quarto"),
-    );
   }

   // record dev config. These are versions as defined in the root configuration file.

 export function pandocBinaryPath(): string {
diff --git a/src/quarto.ts b/src/quarto.ts
index c60382b4e..9c1791ef9 100644
--- a/src/quarto.ts
+++ b/src/quarto.ts
@@ -56,8 +56,7 @@ export async function quarto(
     const installed = readInstalledDevConfig();
     const source = readSourceDevConfig();
     if (installed == null || !devConfigsEqual(installed, source)) {
-      await reconfigureQuarto(installed, source);
-      Deno.exit(1);
+      console.log("Quarto thinks it needs reconfigured");
     }
   }
