diff --git a/src/core/dart-sass.ts b/src/core/dart-sass.ts
index 04764e296..a089d8bcb 100644
--- a/src/core/dart-sass.ts
+++ b/src/core/dart-sass.ts
@@ -15,7 +15,7 @@ import { existsSync } from "fs/mod.ts";
 import { warnOnce } from "./log.ts";
 
 export function dartSassInstallDir() {
-  return toolsPath("dart-sass");
+  return toolsPath("");
 }
 
 export async function dartSassVersion() {
@@ -66,7 +66,7 @@ async function dartCommand(args: string[]) {
     }
 
     const command = Deno.build.os === "windows" ? "sass.bat" : "sass";
-    return toolsPath(join("dart-sass", command));
+    return toolsPath(command);
   };
   const sass = resolvePath();
 
diff --git a/src/core/devconfig.ts b/src/core/devconfig.ts
index bb88c7957..7e5bfa7c2 100644
--- a/src/core/devconfig.ts
+++ b/src/core/devconfig.ts
@@ -67,8 +67,7 @@ export function writeDevConfig(config: DevConfig, binPath: string) {
 }
 
 export function readInstalledDevConfig(): DevConfig | null {
-  const binPath = quartoConfig.binPath();
-  const configPath = join(binPath, "..", "config", kDevConfig);
+  const configPath = join(quartoConfig.basePath(), "package","dist", "config", kDevConfig);
   if (existsSync(configPath)) {
     return JSON.parse(Deno.readTextFileSync(configPath)) as DevConfig;
   } else {
diff --git a/src/core/quarto.ts b/src/core/quarto.ts
index 81cb6239c..297a0b8b1 100644
--- a/src/core/quarto.ts
+++ b/src/core/quarto.ts
@@ -26,6 +26,7 @@ export interface QuartoConfig {
 let dotenvConfig: DotenvConfig;
 
 export const quartoConfig = {
+  basePath: () => getenv("QUARTO_BASE_PATH"),
   binPath: () => getenv("QUARTO_BIN_PATH"),
   toolsPath: () => join(getenv("QUARTO_BIN_PATH"), "tools"),
   sharePath: () => getenv("QUARTO_SHARE_PATH"),
@@ -67,7 +68,7 @@ export function monitorPreviewTerminationConditions(cleanup?: VoidFunction) {
   // src code change
   if (quartoConfig.isDebug()) {
     const srcDir = Deno.realPathSync(
-      join(quartoConfig.binPath(), "../../../src"),
+      join(quartoConfig.basePath(), "src"),
     );
     const watcher = Deno.watchFs([srcDir], { recursive: true });
     const watchForChanges = async () => {
diff --git a/src/core/resources.ts b/src/core/resources.ts
index 1343f9f2a..87adb772e 100644
--- a/src/core/resources.ts
+++ b/src/core/resources.ts
@@ -54,8 +54,16 @@ export function toolsPath(binary: string): string {
       }
     }
   }
-
-  return join(quartoConfig.toolsPath(), binary);
+  if (binary === "pandoc"){
+    const pandoc    = join(quartoConfig.binPath(),binary);
+    const pandocBin = pandoc.concat("-bin");
+	 if(!existsSync(pandoc) && existsSync(pandocBin)){
+      return pandocBin;
+    }else{
+      return pandoc;
+    }
+  }
+  return join(quartoConfig.binPath(), binary);
 }
 
 export function pandocBinaryPath(): string {
