diff -ruN quarto-cli-0.9.256.orig/src/core/dart-sass.ts quarto-cli-0.9.256/src/core/dart-sass.ts
--- quarto-cli-0.9.256.orig/src/core/dart-sass.ts	2022-04-15 10:46:41.000000000
+++ quarto-cli-0.9.256/src/core/dart-sass.ts	2022-04-15 15:36:21.900802245
@@ -6,13 +6,13 @@
 */
 import { join } from "path/mod.ts";
 
-import { toolsPath } from "./resources.ts";
+import { binaryPath } from "./resources.ts";
 import { execProcess } from "./process.ts";
 import { TempContext } from "./temp.ts";
 import { lines } from "./text.ts";
 
 export function dartSassInstallDir() {
-  return toolsPath("dart-sass");
+  return binaryPath("");
 }
 
 export async function dartSassVersion() {
@@ -51,7 +51,7 @@
 
 async function dartCommand(args: string[]) {
   const command = Deno.build.os === "windows" ? "sass.bat" : "sass";
-  const sass = toolsPath(join("dart-sass", command));
+  const sass = binaryPath(command);
   const cmd = [
     sass,
     ...args,
diff -ruN quarto-cli-0.9.256.orig/src/core/devconfig.ts quarto-cli-0.9.256/src/core/devconfig.ts
--- quarto-cli-0.9.256.orig/src/core/devconfig.ts	2022-04-15 10:46:41.000000000
+++ quarto-cli-0.9.256/src/core/devconfig.ts	2022-04-15 15:36:21.901802233
@@ -54,8 +54,7 @@
 }
 
 export function readInstalledDevConfig(): DevConfig | null {
-  const binPath = quartoConfig.binPath();
-  const configPath = join(binPath, "..", "config", kDevConfig);
+  const configPath = join(quartoConfig.basePath(), "package","dist", "config", kDevConfig);
   if (existsSync(configPath)) {
     return JSON.parse(Deno.readTextFileSync(configPath)) as DevConfig;
   } else {
@@ -81,13 +80,7 @@
     }
   };
 
-  const srcScriptPath = join(
-    quartoConfig.binPath(),
-    "..",
-    "..",
-    "scripts",
-    isWindows() ? "windows" : "common",
-  );
+  const srcScriptPath = quartoConfig.binPath();
   return createDevConfig(
     readConfig("DENO"),
     readConfig("DENO_DOM"),
diff -ruN quarto-cli-0.9.256.orig/src/core/esbuild.ts quarto-cli-0.9.256/src/core/esbuild.ts
--- quarto-cli-0.9.256.orig/src/core/esbuild.ts	2022-04-15 10:46:41.000000000
+++ quarto-cli-0.9.256/src/core/esbuild.ts	2022-04-15 15:48:55.566380129
@@ -6,7 +6,7 @@
 */
 
 import { execProcess } from "./process.ts";
-import { toolsPath } from "./resources.ts";
+import { binaryPath } from "./resources.ts";
 
 export async function esbuildCompile(
   input: string,
@@ -30,7 +30,7 @@
   workingDir: string,
 ) {
   const cmd = [
-    toolsPath("esbuild"),
+    binaryPath("esbuild"),
     ...args,
   ];
 
diff -ruN quarto-cli-0.9.256.orig/src/core/quarto.ts quarto-cli-0.9.256/src/core/quarto.ts
--- quarto-cli-0.9.256.orig/src/core/quarto.ts	2022-04-15 10:46:41.000000000
+++ quarto-cli-0.9.256/src/core/quarto.ts	2022-04-15 15:36:21.901802233
@@ -18,6 +18,7 @@
 }
 
 export const quartoConfig = {
+  basePath: () => getenv("QUARTO_BASE_PATH"),
   binPath: () => getenv("QUARTO_BIN_PATH"),
   toolsPath: () => join(getenv("QUARTO_BIN_PATH"), "tools"),
   sharePath: () => getenv("QUARTO_SHARE_PATH"),
diff -ruN quarto-cli-0.9.256.orig/src/core/resources.ts quarto-cli-0.9.256/src/core/resources.ts
--- quarto-cli-0.9.256.orig/src/core/resources.ts	2022-04-15 10:46:41.000000000
+++ quarto-cli-0.9.256/src/core/resources.ts	2022-04-15 15:38:24.458270066
@@ -29,6 +29,10 @@
   return join(resourcePath("formats"), format, resource);
 }
 
+export function binaryPath(binary: string): string {
+  return join(quartoConfig.binPath(), binary);
+}
+
 export function toolsPath(binary: string): string {
   return join(quartoConfig.toolsPath(), binary);
 }
@@ -47,7 +51,7 @@
     }
   }
 
-  return toolsPath("pandoc");
+  return binaryPath("pandoc");
 }
 
 export async function rBinaryPath(binary: string): Promise<string> {
diff -ruN quarto-cli-0.9.256.orig/src/core/run/deno.ts quarto-cli-0.9.256/src/core/run/deno.ts
--- quarto-cli-0.9.256.orig/src/core/run/deno.ts	2022-04-15 10:46:41.000000000
+++ quarto-cli-0.9.256/src/core/run/deno.ts	2022-04-15 15:52:02.589042024
@@ -10,7 +10,7 @@
 import { extname, join } from "path/mod.ts";
 import { quartoCacheDir } from "../appdirs.ts";
 import { execProcess } from "../process.ts";
-import { resourcePath, toolsPath } from "../resources.ts";
+import { resourcePath, binaryPath } from "../resources.ts";
 import { RunHandler, RunHandlerOptions } from "./run.ts";
 import { removeIfExists } from "../path.ts";
 
@@ -37,7 +37,7 @@
 
     return await execProcess({
       cmd: [
-        toolsPath("deno"),
+        binaryPath("deno"),
         "run",
         "--import-map",
         importMap,
diff -ruN quarto-cli-0.9.256.orig/src/quarto.ts quarto-cli-0.9.256/src/quarto.ts
--- quarto-cli-0.9.256.orig/src/quarto.ts	2022-04-15 10:46:41.000000000
+++ quarto-cli-0.9.256/src/quarto.ts	2022-04-15 15:36:21.902802220
@@ -42,8 +42,7 @@
     const installed = readInstalledDevConfig();
     const source = readSourceDevConfig();
     if (installed == null || !devConfigsEqual(installed, source)) {
-      await reconfigureQuarto(installed, source);
-      Deno.exit(1);
+      console.log("Quarto thinks it needs reconfigured");
     }
   }
 
