diff --git a/src/core/dart-sass.ts b/src/core/dart-sass.ts
index 82fc441..62cacdf 100644
--- a/src/core/dart-sass.ts
+++ b/src/core/dart-sass.ts
@@ -6,13 +6,13 @@
 */
 import { join } from "path/mod.ts";
 
-import { toolsPath } from "./resources.ts";
+import { binaryPath } from "./resources.ts";
 import { execProcess } from "./process.ts";
 import { TempContext } from "./temp.ts";
 import { lines } from "./text.ts";
 
 export function dartSassInstallDir() {
-  return toolsPath("dart-sass");
+  return binaryPath("");
 }
 
 export async function dartSassVersion() {
@@ -51,7 +51,7 @@ export async function dartCompile(
 
 async function dartCommand(args: string[]) {
   const command = Deno.build.os === "windows" ? "sass.bat" : "sass";
-  const sass = toolsPath(join("dart-sass", command));
+  const sass = binaryPath(command);
   const cmd = [
     sass,
     ...args,
diff --git a/src/core/devconfig.ts b/src/core/devconfig.ts
index 4ee7e26..db6abf5 100644
--- a/src/core/devconfig.ts
+++ b/src/core/devconfig.ts
@@ -45,12 +45,12 @@ export function createDevConfig(
     script: md5Hash(Deno.readTextFileSync(scriptPath)),
     importMap: md5Hash(
       Deno.readTextFileSync(
-        join(scriptPath, "../../../../src/import_map.json"),
+        join(quartoConfig.basePath(), "/src/import_map.json"),
       ),
     ),
     bundleImportMap: md5Hash(
       Deno.readTextFileSync(
-        join(scriptPath, "../../../../src/resources/vendor/import_map.json"),
+        join(quartoConfig.basePath(), "/src/resources/vendor/import_map.json"),
       ),
     ),
   };
@@ -66,8 +66,7 @@ export function writeDevConfig(config: DevConfig, binPath: string) {
 }
 
 export function readInstalledDevConfig(): DevConfig | null {
-  const binPath = quartoConfig.binPath();
-  const configPath = join(binPath, "..", "config", kDevConfig);
+  const configPath = join(quartoConfig.basePath(), "package","dist", "config", kDevConfig);
   if (existsSync(configPath)) {
     return JSON.parse(Deno.readTextFileSync(configPath)) as DevConfig;
   } else {
@@ -93,13 +92,7 @@ export function readSourceDevConfig(): DevConfig {
     }
   };
 
-  const srcScriptPath = join(
-    quartoConfig.binPath(),
-    "..",
-    "..",
-    "scripts",
-    isWindows() ? "windows" : "common",
-  );
+  const srcScriptPath = quartoConfig.binPath();
   return createDevConfig(
     readConfig("DENO"),
     readConfig("DENO_DOM"),
diff --git a/src/core/esbuild.ts b/src/core/esbuild.ts
index a7b4406..d26bf33 100644
--- a/src/core/esbuild.ts
+++ b/src/core/esbuild.ts
@@ -6,7 +6,7 @@
 */
 
 import { execProcess } from "./process.ts";
-import { toolsPath } from "./resources.ts";
+import { binaryPath } from "./resources.ts";
 
 export async function esbuildCompile(
   input: string,
@@ -30,7 +30,7 @@ export async function esbuildCommand(
   workingDir: string,
 ) {
   const cmd = [
-    toolsPath("esbuild"),
+    binaryPath("esbuild"),
     ...args,
   ];
 
diff --git a/src/core/quarto.ts b/src/core/quarto.ts
index 1f987e8..dc0a8d4 100644
--- a/src/core/quarto.ts
+++ b/src/core/quarto.ts
@@ -24,6 +24,7 @@ export interface QuartoConfig {
 let dotenvConfig: DotenvConfig;
 
 export const quartoConfig = {
+  basePath: () => getenv("QUARTO_BASE_PATH"),
   binPath: () => getenv("QUARTO_BIN_PATH"),
   toolsPath: () => join(getenv("QUARTO_BIN_PATH"), "tools"),
   sharePath: () => getenv("QUARTO_SHARE_PATH"),
@@ -55,7 +56,7 @@ export const quartoConfig = {
 export function monitorQuartoSrcChanges(cleanup?: VoidFunction) {
   if (quartoConfig.isDebug()) {
     const srcDir = Deno.realPathSync(
-      join(quartoConfig.binPath(), "../../../src"),
+      join(quartoConfig.basePath(), "src"),
     );
     const watcher = Deno.watchFs([srcDir], { recursive: true });
     const watchForChanges = async () => {
diff --git a/src/core/resources.ts b/src/core/resources.ts
index fbb81a4..8c5474c 100644
--- a/src/core/resources.ts
+++ b/src/core/resources.ts
@@ -29,6 +29,10 @@ export function formatResourcePath(format: string, resource: string) {
   return join(resourcePath("formats"), format, resource);
 }
 
+export function binaryPath(binary: string): string {
+  return join(quartoConfig.binPath(), binary);
+}
+
 export function toolsPath(binary: string): string {
   return join(quartoConfig.toolsPath(), binary);
 }
@@ -47,7 +51,7 @@ export function pandocBinaryPath(): string {
     }
   }
 
-  return toolsPath("pandoc");
+  return binaryPath("pandoc");
 }
 
 export async function rBinaryPath(binary: string): Promise<string> {
diff --git a/src/core/run/deno.ts b/src/core/run/deno.ts
index 2628750..b34a575 100644
--- a/src/core/run/deno.ts
+++ b/src/core/run/deno.ts
@@ -9,7 +9,7 @@ import { existsSync } from "fs/mod.ts";
 import { extname, join } from "path/mod.ts";
 import { quartoCacheDir } from "../appdirs.ts";
 import { execProcess } from "../process.ts";
-import { resourcePath, toolsPath } from "../resources.ts";
+import { resourcePath, binaryPath } from "../resources.ts";
 import { RunHandler, RunHandlerOptions } from "./types.ts";
 import { removeIfExists } from "../path.ts";
 import { copyTo } from "../copy.ts";
@@ -37,7 +37,7 @@ export const denoRunHandler: RunHandler = {
 
     return await execProcess({
       cmd: [
-        toolsPath("deno"),
+        binaryPath("deno"),
         "run",
         "--import-map",
         importMap,
diff --git a/src/quarto.ts b/src/quarto.ts
index 1e5c618..1451fc1 100644
--- a/src/quarto.ts
+++ b/src/quarto.ts
@@ -52,8 +52,7 @@ export async function quarto(
     const installed = readInstalledDevConfig();
     const source = readSourceDevConfig();
     if (installed == null || !devConfigsEqual(installed, source)) {
-      await reconfigureQuarto(installed, source);
-      Deno.exit(1);
+      console.log("Quarto thinks it needs reconfigured");
     }
   }
 
