From a69551336f37cd4010032e039231d926e1a4774c Mon Sep 17 00:00:00 2001
From: b-fuze <b-fuze.root@live.com>
Date: Thu, 25 Aug 2022 21:07:48 -0400
Subject: [PATCH] fix(native): pointer -> buffer

---
 deno-dom-native.ts | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/deno-dom-native.ts b/deno-dom-native.ts
index 879b384..eec77f3 100644
--- a/deno-dom-native.ts
+++ b/deno-dom-native.ts
@@ -14,15 +14,15 @@ type DeepWriteable<T> = { -readonly [P in keyof T]: DeepWriteable<T[P]> };
 const _symbols = {
   deno_dom_usize_len: { parameters: [], result: "usize" },
   deno_dom_parse_sync: {
-    parameters: ["pointer", "usize", "pointer"],
+    parameters: ["buffer", "usize", "buffer"],
     result: "void",
   },
   deno_dom_parse_frag_sync: {
-    parameters: ["pointer", "usize", "pointer"],
+    parameters: ["buffer", "usize", "buffer"],
     result: "void",
   },
   deno_dom_is_big_endian: { parameters: [], result: "u32" },
-  deno_dom_copy_buf: { parameters: ["pointer", "pointer"], result: "void" },
+  deno_dom_copy_buf: { parameters: ["buffer", "buffer"], result: "void" },
 } as const;
 const symbols = _symbols as DeepWriteable<typeof _symbols>;
 
