diff --git a/src/cpp/core/CMakeLists.txt b/src/cpp/core/CMakeLists.txt
index 47e3fb1dd4..b979419eab 100644
--- a/src/cpp/core/CMakeLists.txt
+++ b/src/cpp/core/CMakeLists.txt
@@ -15,7 +15,6 @@
 
 project (CORE)
 
-add_subdirectory(spelling/hunspell)
 add_subdirectory(tex/synctex)
 
 # include files
@@ -325,10 +324,12 @@ include_directories(
 add_library(rstudio-core STATIC ${CORE_SOURCE_FILES} ${CORE_HEADER_FILES})
 define_source_file_names(rstudio-core)
 
+find_library(HUNSPELL_LIBRARIES NAMES hunspell hunspell-1.7 REQUIRED)
+
 # link dependencies
 target_link_libraries(rstudio-core
    rstudio-shared-core
-   rstudio-core-hunspell
+   ${HUNSPELL_LIBRARIES}
    ${Boost_LIBRARIES}
    ${CRASHPAD_LIBRARIES}
    ${SOCI_LIBRARIES}
@@ -355,7 +356,7 @@ if (RSTUDIO_UNIT_TESTS_ENABLED)
       rstudio-shared-core
       rstudio-core
       rstudio-core-synctex
-      rstudio-core-hunspell
+      ${HUNSPELL_LIBRARIES}
       ${Boost_LIBRARIES}
       ${SOCI_LIBRARIES}
       ${CORE_SYSTEM_LIBRARIES}
diff --git a/src/cpp/core/spelling/HunspellSpellingEngine.cpp b/src/cpp/core/spelling/HunspellSpellingEngine.cpp
index f88aba0b07..6951f27d53 100644
--- a/src/cpp/core/spelling/HunspellSpellingEngine.cpp
+++ b/src/cpp/core/spelling/HunspellSpellingEngine.cpp
@@ -33,7 +33,7 @@
 #if defined(near)
 #undef near
 #endif
-#include "hunspell/hunspell.hxx"
+#include <hunspell/hunspell.hxx>
 
 namespace rstudio {
 namespace core {
@@ -195,12 +195,11 @@ public:
 
    Error wordChars(std::wstring *pWordChars)
    {
-      int len;
-      unsigned short *pChars = pHunspell_->get_wordchars_utf16(&len);
-
-      for (int i = 0; i < len; i++)
-         pWordChars->push_back(pChars[i]);
-
+      std::vector<w_char> wChars = pHunspell_->get_wordchars_utf16();
+      for (w_char wChar : wChars){
+         wchar_t char_new = (wChar.h << 8) + wChar.l;
+         pWordChars->push_back(char_new);
+      }
       return Success();
    }
 
diff --git a/src/cpp/session/CMakeLists.txt b/src/cpp/session/CMakeLists.txt
index 571125b57b..d842ceb0fa 100644
--- a/src/cpp/session/CMakeLists.txt
+++ b/src/cpp/session/CMakeLists.txt
@@ -547,7 +546,7 @@ endif()
 
 set(SESSION_LIBRARIES
    rstudio-core
-   rstudio-core-hunspell
+   ${HUNSPELL_LIBRARIES}
    rstudio-core-synctex
    ${RSTUDIO_CORE_ZLIB}
    rstudio-monitor
