--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -197,8 +197,7 @@ endif()
 # yaml-cpp
 set(YAML_CPP_VERSION "0.8.0")
 set(YAML_CPP_REVISION "f7320141120f720aecc4c32be25586e7da9eb978")
-if(CMAKE_VERSION VERSION_GREATER "3.14")
-
+if(CMAKE_VERSION VERSION_GREATER "3.14" AND NOT RSTUDIO_USE_SYSTEM_YAML_CPP)
    if(WIN32)
       add_definitions(-DYAML_CPP_STATIC_DEFINE)
    endif()
@@ -237,13 +236,6 @@ if(YAML_CPP_INCLUDE_DIR)
    include_directories(SYSTEM "${YAML_CPP_INCLUDE_DIR}")
 endif()
 
-# determine whether we should statically link boost. we always do this
-# unless we are building a non-packaged build on linux (in which case
-# boost dynamic libraries are presumed to be installed on the system ldpath)
-if(APPLE OR WIN32 OR RSTUDIO_PACKAGE_BUILD)
-   set(Boost_USE_STATIC_LIBS ON)
-endif()
-
 # default Boost versions
 if(NOT RSTUDIO_BOOST_REQUESTED_VERSION)
    set(RSTUDIO_BOOST_REQUESTED_VERSION 1.83.0)
--- a/src/cpp/desktop/CMakeLists.txt
+++ b/src/cpp/desktop/CMakeLists.txt
@@ -328,18 +328,6 @@ if(NOT RSTUDIO_ELECTRON)
    endif(WIN32)
 
 
-   # determine whether we should bundle Qt. we pretty much always want to
-   # bundle it unless we are building on linux and NOT linking against the
-   # qtsdk -- in that case we are linking against system versions of the
-   # qt libraries which should therefore not be bundled
-   if(APPLE OR WIN32 OR
-      RSTUDIO_PACKAGE_BUILD OR
-      (${QT_LIBRARY_DIR} MATCHES ".*QtSDK.*"))
-
-      set(RSTUDIO_BUNDLE_QT TRUE)
-
-   endif()
-
    set(RSTUDIO_QTMODULES
       Qt5::Core
       Qt5::Widgets
