--- a/cmake/globals.cmake
+++ b/cmake/globals.cmake
@@ -319,6 +319,10 @@ endif()
 
 message(STATUS "Using RStudio tools root: ${RSTUDIO_TOOLS_ROOT}")
 
+if(RSTUDIO_INSTALL_SUPPORTING)
+   add_compile_definitions(RSTUDIO_RESOURCES_DIR="${RSTUDIO_INSTALL_SUPPORTING}")
+endif()
+
 # special install directories for apple desktop
 if (APPLE)
    if (RSTUDIO_ELECTRON)
@@ -335,17 +339,30 @@ else()
       if (RSTUDIO_SESSION_WIN32)
          set(RSTUDIO_INSTALL_BIN resources/app/bin/x86)
       else()
-         set(RSTUDIO_INSTALL_BIN resources/app/bin)
+         if(NOT RSTUDIO_INSTALL_SUPPORTING)
+            set(RSTUDIO_INSTALL_BIN resources/app/bin)
+         else()
+            set(RSTUDIO_INSTALL_BIN ${RSTUDIO_INSTALL_SUPPORTING}/resources/app/bin)
+         endif()
+
       endif()
-      set(RSTUDIO_INSTALL_SUPPORTING resources/app)
-      set(RSTUDIO_INSTALL_ELECTRON .)
+      if(NOT RSTUDIO_INSTALL_SUPPORTING)
+	 set(RSTUDIO_INSTALL_SUPPORTING resources/app)
+        set(RSTUDIO_INSTALL_ELECTRON .)
+      else()
+	 #set(RSTUDIO_INSTALL_SUPPORTING ${RSTUDIO_INSTALL_SUPPORTING})
+        set(RSTUDIO_INSTALL_ELECTRON ${RSTUDIO_INSTALL_SUPPORTING})
+      endif()
+
    else()
       if (RSTUDIO_SESSION_WIN32)
          set(RSTUDIO_INSTALL_BIN x86)
       else()
          set(RSTUDIO_INSTALL_BIN bin)
       endif()
-      set(RSTUDIO_INSTALL_SUPPORTING .)
+      if(NOT RSTUDIO_INSTALL_SUPPORTING)
+         set(RSTUDIO_INSTALL_SUPPORTING .)
+      endif()
    endif()
 endif()
 
--- a/src/cpp/server/ServerOptions.cpp
+++ b/src/cpp/server/ServerOptions.cpp
@@ -164,6 +164,15 @@ ProgramStatus Options::read(int argc,
    }
 #endif
 
+#ifdef RSTUDIO_RESOURCES_DIR
+   error = core::system::realPath(RSTUDIO_RESOURCES_DIR, &resourcePath);
+   if (error)
+   {
+      LOG_ERROR_MESSAGE("Unable to determine resource path: "+error.getSummary());
+      return ProgramStatus::exitFailure();
+   }
+
+#endif
    // special program offline option (based on file existence at 
    // startup for easy bash script enable/disable of offline state)
    serverOffline_ = FilePath("/var/lib/rstudio-server/offline").exists();
--- a/src/cpp/server_core/DatabaseUtils.cpp
+++ b/src/cpp/server_core/DatabaseUtils.cpp
@@ -436,8 +436,14 @@ Error migrationsDir(const std::string& rootDir, FilePath* pMigrationsDir)
 
 Error migrationsDir(const std::string& rootDir, const std::string& migrationEnvVarName, core::FilePath* pMigrationsDir)
 {
+
+#ifdef RSTUDIO_RESOURCES_DIR
+   FilePath resourcePath;
+   Error error = core::system::realPath(RSTUDIO_RESOURCES_DIR, &resourcePath);
+#else
    FilePath exePath;
    Error error = core::system::executablePath(nullptr, &exePath);
+#endif
    if (error)
       return error;
 
@@ -453,8 +459,11 @@ Error migrationsDir(const std::string& rootDir, const std::string& migrationEnvV
          return Success();
       }
    }
-   
+#ifdef RSTUDIO_RESOURCES_DIR
+   *pMigrationsDir = resourcePath.completeChildPath(rootDir);
+#else
    *pMigrationsDir = exePath.getParent().getParent().completeChildPath(rootDir);
+#endif
 
    return Success();
 }
--- a/src/cpp/session/SessionOptions.cpp
+++ b/src/cpp/session/SessionOptions.cpp
@@ -105,7 +105,11 @@ core::ProgramStatus Options::read(int argc, char * const argv[], std::ostream& o
    core::system::unsetenv(kMonitorSharedSecretEnvVar);
 
    // compute the resource path
+#ifdef RSTUDIO_RESOURCES_DIR
+   Error error = core::system::realPath(RSTUDIO_RESOURCES_DIR, &resourcePath_);
+#else
    Error error = core::system::installPath("..", argv[0], &resourcePath_);
+#endif
    if (error)
    {
       LOG_ERROR_MESSAGE("Unable to determine install path: " + error.getSummary());
