diff --git a/src/node/desktop/CMakeLists.txt b/src/node/desktop/CMakeLists.txt
index da724c6e1f..b6ebbc4519 100644
--- a/src/node/desktop/CMakeLists.txt
+++ b/src/node/desktop/CMakeLists.txt
@@ -167,6 +168,8 @@ else()
          ${CMAKE_COMMAND} -E env
          "PATH=${MODIFIED_PATH}"
          "MAKEFLAGS="
+	 "HOME=$ENV{WORKDIR}"
+	 "XDG_CACHE_HOME=$ENV{WORKDIR}/.cache"
          ${NPM} run package
    )
 
