diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index d210616..7788409 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -268,7 +268,7 @@ if(UNIX)
       set(Boost_USE_STATIC_LIBS ON)
       set(BOOST_INCLUDEDIR  ${RSTUDIO_TOOLS_BOOST}/include)
       find_package(Boost ${RSTUDIO_BOOST_REQUESTED_VERSION} EXACT REQUIRED)
-      if(NOT Boost_VERSION LESS 106900)
+      if(NOT Boost_VERSION LESS 1.69.00)
          list(REMOVE_ITEM BOOST_LIBS signals)
       endif()
 
@@ -281,7 +281,7 @@ if(UNIX)
    else()
       add_definitions(-DRSTUDIO_BOOST_NAMESPACE=boost)
       find_package(Boost ${RSTUDIO_BOOST_REQUESTED_VERSION} REQUIRED)
-      if(NOT Boost_VERSION LESS 106900)
+      if(NOT Boost_VERSION LESS 1.69.00)
          list(REMOVE_ITEM BOOST_LIBS signals)
       endif()
       find_package(Boost ${RSTUDIO_BOOST_REQUESTED_VERSION} REQUIRED COMPONENTS ${BOOST_LIBS})
