
diff -ur rstudio-2021.09.1-372/src/cpp/CMakeLists.txt rstudio-2021.09.1-372_new/src/cpp/CMakeLists.txt
--- rstudio-2021.09.1-372/src/cpp/CMakeLists.txt	2021-11-08 17:04:03.000000000
+++ rstudio-2021.09.1-372_new/src/cpp/CMakeLists.txt	2021-12-30 10:24:33.487951577
@@ -224,13 +224,6 @@
 
 include_directories(SYSTEM "${YAML_CPP_INCLUDE_DIR}")
 
-# determine whether we should statically link boost. we always do this
-# unless we are building a non-packaged build on linux (in which case
-# boost dynamic libraries are presumed to be installed on the system ldpath)
-if(APPLE OR WIN32 OR RSTUDIO_PACKAGE_BUILD)
-   set(Boost_USE_STATIC_LIBS ON)
-endif()
-
 # default Boost versions
 if(NOT RSTUDIO_BOOST_REQUESTED_VERSION)
    set(RSTUDIO_BOOST_REQUESTED_VERSION 1.69.0)

diff -ur rstudio-2021.09.1-372/src/cpp/desktop/CMakeLists.txt rstudio-2021.09.1-372_new/src/cpp/desktop/CMakeLists.txt
--- rstudio-2021.09.1-372/src/cpp/desktop/CMakeLists.txt	2021-11-08 17:04:03.000000000
+++ rstudio-2021.09.1-372_new/src/cpp/desktop/CMakeLists.txt	2021-12-30 13:47:00.244096228
@@ -318,16 +318,6 @@
 endif(WIN32)
 
 
-# determine whether we should bundle Qt. we pretty much always want to
-# bundle it unless we are building on linux and NOT linking against the
-# qtsdk -- in that case we are linking against system versions of the
-# qt libraries which should therefore not be bundled
-if(APPLE OR WIN32 OR
-   RSTUDIO_PACKAGE_BUILD OR
-   (${QT_LIBRARY_DIR} MATCHES ".*QtSDK.*"))
-  set(RSTUDIO_BUNDLE_QT TRUE)
-endif()
-
 set(RSTUDIO_QTMODULES
     Qt5::Core
     Qt5::Widgets

