diff --git a/CMakeGlobals.txt b/CMakeGlobals.txt
index 025efb6279..2e29dd12a9 100644
--- a/CMakeGlobals.txt
+++ b/CMakeGlobals.txt
@@ -232,6 +232,7 @@ endif()
 
 # pandoc version
 set(PANDOC_VERSION "2.18" CACHE INTERNAL "Pandoc version")
+add_compile_definitions(RSTUDIO_PANDOC_PATH="${RSTUDIO_PANDOC_PATH}")
 
 # quarto support
 if(LINUX AND UNAME_M STREQUAL aarch64)
diff --git a/src/cpp/session/include/session/SessionConstants.hpp b/src/cpp/session/include/session/SessionConstants.hpp
index ce49a90bcb..4b3b652f86 100644
--- a/src/cpp/session/include/session/SessionConstants.hpp
+++ b/src/cpp/session/include/session/SessionConstants.hpp
@@ -138,11 +138,7 @@
 #define kSessionTmpDirEnvVar       "RS_SESSION_TMP_DIR"
 #define kSessionTmpDir             "rstudio-rsession"
 
-#ifdef QUARTO_ENABLED
-# define kDefaultPandocPath        "bin/quarto/bin/tools"
-#else
-# define kDefaultPandocPath        "bin/pandoc"
-#endif
+#define kDefaultPandocPath         RSTUDIO_PANDOC_PATH
 
 #define kDefaultQuartoPath         "bin/quarto"
 #define kDefaultPostbackPath       "bin/postback/rpostback"
